spec/poisson_spec.rb

```diff
@@ -1,6 +1,36 @@
 require File.expand_path(File.dirname(__FILE__) + '/spec_helper.rb')
 include ExampleWithGSL
 describe Distribution::Poisson do
+  shared_examples_for 'poisson engine(with rng)' do
+    it 'should return correct rng' do
+      lambda_val = 2
+      exp_mean = 2
+      exp_seed = 1
+
+      # if seed = "1" / lambda_val = 2, rng is expected to all "2"
+      rng = @engine.rng(lambda_val, exp_seed)
+
+      samples = 100
+      sum = 0
+      samples.times do
+        v = rng.call
+        sum += v
+      end
+      mean = sum / samples
+      expect(mean.to_i).to 2
```

Comment:
Prefer using the `be_within` matcher for floating point comparisons:

```suggestion
      expect(mean.to_i).to be_within(ERROR_DELTA).of(2)
```

Reply:
As for an `ERROR_DELTA` value, I'd recommend picking the smallest power of `10` that works without slowing down the tests too much would be great. My rationale for the suggestion is to document a soft guarantee in the tests. I don't have a strong opinion about this, but anything larger than `1e-4` would be too little precision. Perhaps other maintainers who look at this PR can chip in about that?

Reply:
I had change the code to use `be_within()`.
And I picked the value of `1e-10` for `ERROR_DELTA`, because there was used the value of `1e-10` in another test, `normal_spec.rb`, `binomial_spec.rb` etc... 
About the value of `ERROR_DELTA`, I'd decided to learn from the previous example.